app/features/MidiData.py

```diff
@@ -0,0 +1,50 @@
+from mido import MidiFile, tempo2bpm, tick2second, merge_tracks
+# from app.features.Type.Notes import Notes
+# from app.features.Type.Tempos import Tempos
+from app.features.Type.Notes import Notes
+from app.features.Type.Tempos import Tempos
+
+
+class MidiData:
+    def __init__(self, file_name):
+        self.mid = MidiFile(file_name)
+        self.tempos = Tempos()
+        self.notes = Notes()
+
+        self.parseData()
+
+    def parseData(self):
+        print("Data recovery...")
+
+        self.notes.setTime(self.mid.length)
+        self.tempos.setTime(self.mid.length)
+        currentTime = 0
+        tempoTemp = 0
+        for msg in merge_tracks(self.mid.tracks):
+            currentTime = tick2second(msg.time, self.mid.ticks_per_beat, tempoTemp) + currentTime
+            if msg.type == 'set_tempo':
+                tempoTemp = msg.tempo
+                # fill tempo
+                self.tempos.tempos.append({'tempo': round(tempo2bpm(msg.tempo)), 'currentTime': currentTime})
+                # fill tempo with repetition
+                find = False
+                for j in range(len(self.tempos.temposWithRepetition)):
+                    if self.tempos.temposWithRepetition[j].get('tempo') == round(tempo2bpm(msg.tempo)):
+                        self.tempos.temposWithRepetition[j]['repetition'] = self.tempos.temposWithRepetition[j][
+                                                                                  'repetition'] + 1
+                        self.tempos.temposWithRepetition[j]['time'] = tick2second(msg.time, self.mid.ticks_per_beat, msg.tempo) + self.tempos.temposWithRepetition[j]['time']
+                        find = True
+                if not find:
+                    self.tempos.temposWithRepetition.append(
+                        {'tempo': round(tempo2bpm(msg.tempo)), 'repetition': 1, 'time': tick2second(msg.time, self.mid.ticks_per_beat, msg.tempo)})
```

Comment:
Should this go inside the `Tempos` class ?
The class would directly have a method `append` that receives the tempo `msg` and` currentTime` and does this job of finding the repetitions, etc..